pkg/controller/backup_deletion_controller.go

```diff
@@ -295,34 +295,39 @@ func (c *backupDeletionController) processRequest(req *velerov1api.DeleteBackupR
 		errs = append(errs, err.Error())
 	}
 
-	// Download the tarball
-	backupFile, err := downloadToTempFile(backup.Name, backupStore, log)
-	if err != nil {
-		return errors.Wrap(err, "error downloading backup")
-	}
-	defer closeAndRemoveFile(backupFile, c.logger)
-
 	actions, err := pluginManager.GetDeleteItemActions()
 	log.Debugf("%d actions before invoking actions", len(actions))
 	if err != nil {
 		return errors.Wrap(err, "error getting delete item actions")
 	}
 	// don't defer CleanupClients here, since it was already called above.
 
-	ctx := &delete.Context{
-		Backup:          backup,
-		BackupReader:    backupFile,
-		Actions:         actions,
-		Log:             c.logger,
-		DiscoveryHelper: c.helper,
-		Filesystem:      filesystem.NewFileSystem(),
-	}
+	if len(actions) > 0 {
+		// Download the tarball
+		backupFile, err := downloadToTempFile(backup.Name, backupStore, log)
+
+		if err != nil {
+			log.WithError(err).Errorf("Unable to download tarball for backup %s, skipping associated DeleteItemAction plugins", backup.Name)
```

Comment:
Does this approach seem okay? Unfortunately we can't distinguish the reasons why downloading the backup tarball fails. It could be missing, or it could be a temporary issue. With this approach, if it ever fails we will always skip the plugins and proceed with deleting the rest of the backup resources.

Reply:
I think some retry logic in the backupStore.GetBackupContents function would be helpful for sure, so we can account for the transient failures better.

I don't know if it needs to flow down to the ObjectStore plugin interface though - if so, I could see it growing at least a couple new methods and settings (number of retries, max timeout, whether or not to do exponential backoff, etc). I'd have to think more about that, but I think a plugin-level change isn't right for this PR.

Reply:
Another option is to extend our error types to help us distinguish, though this would likely also require participation by plugin authors to return the right ones.

Kubernetes has this concept, and we use it in various places, such as [backup.go](https://github.com/vmware-tanzu/velero/blob/main/pkg/backup/backup.go#L490)

Reply:
Hmm, I've been testing this further and from my local setup, the error doesn't come from `backupStore.GetBackupContents` but actually comes from trying to read from the resulting `io.ReadCloser` returned by `GetBackupContents`. I had added some retry logic to that function but couldn't get it to trigger. I don't know if it's worth keeping it in.

Reply:
Interesting - so you get an `io.ReadCloser`, but it's not valid? Is it `nil` when that happens?

We might just have to revisit the retry logic separately for now.

Reply:
It's a valid `io.ReadCloser` but it's a [`StreamReadCloser`](https://github.com/vmware-tanzu/velero/blob/main/pkg/plugin/framework/stream_reader.go#L34-L40). `GetBackupContents` eventually calls the grpc `GetObject` method and it [returns a stream](https://github.com/vmware-tanzu/velero/blob/main/pkg/plugin/proto/ObjectStore.proto#L81) which we don't attempt to read from until we call [`io.Copy`](https://github.com/vmware-tanzu/velero/blob/main/pkg/controller/restore_controller.go#L556) later in the [`downloadToTempFile` function](https://github.com/vmware-tanzu/velero/blob/main/pkg/controller/restore_controller.go#L556).

I initially thought this might have been a plugin specific implementation issue, but it seems to be the case across two object store plugins that I've checked.  I'll revert the retry logic for now.